src/Microsoft.Extensions.DependencyInjection/ServiceCollectionContainerBuilderExtensions.cs

```diff
@@ -9,7 +9,12 @@ public static class ServiceCollectionContainerBuilderExtensions
     {
         public static IServiceProvider BuildServiceProvider(this IServiceCollection services)
         {
-            return new ServiceProvider(services);
+            return BuildServiceProvider(services, validateScopes: false);
+        }
+
+        public static IServiceProvider BuildServiceProvider(this IServiceCollection services, bool validateScopes)
```

Comment:
Time for doc comments? On one hand, I kind of hope no one ever uses this (except for may us). On the other hand, without context, I would have no idea what `validateScopes` is even about.